app/metrics/views/model_views.py

```diff
@@ -127,6 +128,107 @@ def get_stats(self):
             *field_stats
         ]
 
+class TessImportEventView(LoginRequiredMixin, CreateView):
+    # Figure out how to avoid duplicating this from EventView
+    model = models.Event
+    fields = [
+        "title",
+        "node",
+        "date_start",
+        "date_end",
+        "duration",
+        "type",
+        "organising_institution",
+        "location_city",
+        "location_country",
+        "funding",
+        "target_audience",
+        "additional_platforms",
+        "communities",
+        "number_participants",
+        "number_trainers",
+        "url",
+        "status",
+    ]
+    template_name = "metrics/tess-import-form.html"
+    tess_id = None
+    title = "Import from TeSS"
+    tess_metadata = {}
+    tess_url = None
+
+    def get_actions(self):
+        return []
+
+    def get_stats(self):
+        return None
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["title"] = self.title
+        for field in context["form"]:
+            field.field.widget.attrs.update({
+                "class": "form-control",
+            })
+        context["actions"] = self.get_actions()
+        context["stats"] = self.get_stats()
+        context.update(get_tabs(self.request))
+        context["can_edit"] = self.can_edit()
+        context["tess_metadata"] = self.tess_metadata
+        context["tess_url"] = self.tess_url
+        return context
+
+    def can_edit(self):
+        return True
+
+    def form_valid(self, form):
+        obj = form.save(commit = False)
+        obj.user = self.request.user
+        obj.node_main = self.request.user.get_node()
+        return super().form_valid(form)
+
+    def get_form(self):
+        tmd_metadata = {}
+        if self.tess_id:
+            tess_metadata = self.import_from_tess(self.tess_id)
+            tmd_metadata = self.convert_tess_metadata(tess_metadata)
+            self.tess_metadata = tess_metadata["data"]["attributes"]
+            self.tess_url = "https://tess.elixir-europe.org" + tess_metadata["data"]["links"]["self"]
+            print(self.tess_url)
+        form = super().get_form()
+        tmd_metadata['user'] = self.request.user
+        tmd_metadata['node_main'] = self.request.user.get_node()
+        for key in form.fields:
+            field = form.fields[key]
+            field.initial = tmd_metadata.get(key, "")
+            field.disabled = not self.can_edit()
+        return form
+
+    def import_from_tess(self, tess_id):
+        tess_url = f"https://tess.elixir-europe.org/events/{tess_id}.json_api"
```

Comment:
Maybe it would make sense to move this fetching to the  `get` function as in overriding https://github.com/django/django/blob/c12bc980e5b2bb25e447cd8dee550cad767f1ad2/django/views/generic/edit.py#L176

And instead of reporting a ValidationError you could report a 404? 

You could maybe set `self.tess_metadata` to make it available for `get_form`.